encode_test.go

```diff
@@ -717,6 +717,20 @@ func TestEncoder(t *testing.T) {
 				yaml.UseSingleQuote(false),
 			},
 		},
+		{
+			// Might be obvious but in case you are confused.
+			//
+			// Here, \ is doubled like \\ in test.source and test.value, only because they are Go string literals.
+			// This is testing that a slash is encoded as-is, without escaping.
+			// In the YAML spec, the only escape sequence is '', which represents a single quote.
+			// \\, \n, \t, or whatever you might expect YAML, Go strings, and Go strconv.Quote() to escape using the extra backslash,
+			// are not escaped and printed as-is in the YAML single-quoted string.
+			"a: '\\.yaml'" + "\n",
```

Comment:
I think this is easier to understand. 
```
`a: \.yaml` + "\n"
```

Also, since this behavior is self-explanatory, I felt that a comment was unnecessary. The current behavior seems to be an obvious bug in the UseSingleQuote option.